src/Closure.cpp

```diff
@@ -38,6 +38,54 @@ void Closure::visit(const For *op) {
     ignore.pop(op->name);
 }
 
+void Closure::visit(const Call *op) {
+    if (op->is_intrinsic(Call::address_of)) {
+        const Load *load = op->args[0].as<Load>();
+        internal_assert(load);
+        if (!ignore.contains(load->name)) {
+            debug(3) << "Adding buffer " << load->name << " to closure\n";
+            Buffer &ref = buffers[load->name];
+            ref.type = load->type.element_of(); // TODO: Validate type is the same as existing refs?
+            ref.read = address_of_read;
+            ref.write = address_of_written;
+
+            // If reading an image/buffer, compute the size.
```

Comment:
Redundant with code in the load and store visitor. Maybe lift it out into a helper method like:

void found_buffer_ref(const string &name, bool read, bool written, Buffer<> image /* Undefined for stores */)